tests/src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -161,6 +263,944 @@ TEST_F(SketchObjectTest, testGeoIdFromShapeTypeRootPoint)
+    // The arcOfConic should be split into two arcs of the same conic now
+    EXPECT_EQ(getObject()->getHighestCurveIndex(), 1);
+    // TODO: Expect the end points of the resultant curve are coincident.
+    int numberOfCoincidentConstraints = countConstraintsOfType(getObject(), Sketcher::Coincident);
+    EXPECT_EQ(numberOfCoincidentConstraints, 1);
+}
+
+TEST_F(SketchObjectTest, testSplitNonPeriodicBSpline)
+{
+    // Arrange
+    auto nonPeriodicBSpline = createTypicalNonPeriodicBSpline();
+    Base::Vector3d splitPoint(-0.5, 1.1, 0.0);
+    int geoId = getObject()->addGeometry(nonPeriodicBSpline.get());
+    // TODO: Put a point on this
+
+    // Act
+    // TODO: sample before point(s) at a random parameter
+    int result = getObject()->split(geoId, splitPoint);
+    for (int iterGeoId = 0; iterGeoId < getObject()->getHighestCurveIndex(); ++iterGeoId) {
+        getObject()->deleteUnusedInternalGeometryAndUpdateGeoId(iterGeoId);
+    }
+
+    // Assert
+    EXPECT_EQ(result, 0);
+    EXPECT_EQ(getObject()->getHighestCurveIndex(), 1);
+    // TODO: confirm sampled point(s) is/are at the same place
+    int numberOfCoincidentConstraints = countConstraintsOfType(getObject(), Sketcher::Coincident);
+    EXPECT_EQ(numberOfCoincidentConstraints, 1);
+}
+
+TEST_F(SketchObjectTest, testSplitPeriodicBSpline)
+{
+    // Arrange
+    auto PeriodicBSpline = createTypicalPeriodicBSpline();
+    Base::Vector3d splitPoint(-0.5, 1.1, 0.0);
+    int geoId = getObject()->addGeometry(PeriodicBSpline.get());
+    // TODO: Put a point on this
+
+    // Act
+    // TODO: sample before point(s) at a random parameter
+    int result = getObject()->split(geoId, splitPoint);
+    for (int iterGeoId = 0; iterGeoId < getObject()->getHighestCurveIndex(); ++iterGeoId) {
+        getObject()->deleteUnusedInternalGeometryAndUpdateGeoId(iterGeoId);
+    }
+
+    // Assert
+    EXPECT_EQ(result, 0);
+    EXPECT_EQ(getObject()->getHighestCurveIndex(), 0);
+    // TODO: confirm sampled point(s) is/are at the same place
+}
+
+TEST_F(SketchObjectTest, testTrimWithoutIntersection)
+{
+    // Arrange
+    Part::GeomLineSegment lineSeg;
+    setupLineSegment(lineSeg);
+    int geoId = getObject()->addGeometry(&lineSeg);
+    Base::Vector3d trimPoint(2.0, 3.1, 0.0);
+
+    // Act
+    int result = getObject()->trim(geoId, trimPoint);
+
+    // Assert
+    EXPECT_EQ(result, 0);
+    // Once this line segment is trimmed, nothing should remain
+    EXPECT_EQ(getObject()->getHighestCurveIndex(), geoId - 1);
+}
+
+// TODO: There are other combinations of constraints we may want to test with trim.
+
+TEST_F(SketchObjectTest, testTrimLineSegmentEnd)
+{
+    // Arrange
+    Part::GeomLineSegment lineSeg;
+    setupLineSegment(lineSeg);
+    // TODO: create curves intersecting at the right spots
+    Base::Vector3d trimPoint(lineSeg.pointAtParameter(
+        lineSeg.getFirstParameter()
+        + (lineSeg.getLastParameter() - lineSeg.getFirstParameter()) * 0.2));
```

Comment:
This and below could use a better formatting. Maybe some function like `getPointAtNormalizedParam(double t)`?

Reply:
Done.